pkg/apis/hive/v1alpha1/dnszone_types.go

```diff
@@ -50,6 +54,16 @@ type AWSResourceTag struct {
 	Value string `json:"value"`
 }
 
+// GCPDNSZoneSpec contains GCP-specific DNSZone specifications
+type GCPDNSZoneSpec struct {
+	// Credentials references a secret that will be used to authenticate with
+	// GCP CloudDNS. It will need permission to create and manage CloudDNS Hosted Zones.
+	// Secret should have a key named 'osServiceAccount.json'.
+	// The credentials must specify the project to use.
+	// +optional
+	Credentials corev1.LocalObjectReference `json:"credentials,omitempty"`
```

Comment:
I want to be consistent here, but we have 2 patterns now.

Above in the `AWSDNSZoneSpec`, it references the secret as `AccountSecret`.

But in the [`PlatformSecrets`](https://github.com/openshift/hive/blob/master/pkg/apis/hive/v1alpha1/gcp/platform.go#L27) and [`ExternalDNSGCPConfig`](https://github.com/openshift/hive/blob/master/pkg/apis/hive/v1alpha1/hiveconfig_types.go#L117) it refers to them as `Credentials`.

I followed the latter pattern here, but is that the correct decision?

Should we fix [`AWSDNSZoneSpec`](https://github.com/openshift/hive/blob/v1/pkg/apis/hive/v1/dnszone_types.go#L31-L42) in v1 to be consistent as well?

Also, not sure we need `Region` in there since Route53 is regionless. Should we remove that for v1 too?

To be clear, I'm not suggesting making those changes in this PR.

@staebler @dgoodwin thoughts?

Reply:
+1 for fixing AWS to be consistent with Credentials.

Reply:
Also this should probably be CredentialsSecretRef in both cases per what we did most recently in v1.

Reply:
@dgoodwin do we want it to be `CredentialsSecretRef` even in the v1alpha1 version?

Reply:
I filed [this bug](https://jira.coreos.com/browse/CO-635) to fix `AWSDNSZoneSpec` in the v1 api.

Reply:
I would yeah, we're inconsistent in v1alpha1 anyhow and it's one less thing we need to change after this merges into v1.

Reply:
Done.